src/media_type_buf.rs

```diff
@@ -21,11 +21,13 @@ pub struct MediaTypeBuf {
 
 impl MediaTypeBuf {
     /// Constructs a `MediaTypeBuf` from a top-level type and a subtype.
+    #[must_use]
     pub fn new(ty: Name, subty: Name) -> Self {
-        Self::from_string(format!("{}/{}", ty, subty)).unwrap()
+        Self::from_string(format!("{}/{}", ty, subty)).expect("`ty` and `subty` should be valid")
```

Comment:
Some messages in `.expect()` are wrong because they are actually unreachable. For example, in this line, `ty` and `subty` are always valid so `expect()` never panics. I think we should use `.expect("unreachable")` here.

Reply:
Fair enough.